tests/tasks/verify_settings.yml

```diff
@@ -0,0 +1,77 @@
+- name: Verify edition
+  when: __verify_mssql_edition is defined
+  block:
+    - name: Check if the errorlog file exists and its location
+      shell: |
+        set -euo pipefail
+        errorlog="$(/opt/mssql/bin/mssql-conf get filelocation errorlogfile \
+        2>&1 | sed 's\errorlogfile : \\')" || :
+        if [ -f "${errorlog}" ]; then
+          echo "${errorlog}"
+        elif [ -f /var/opt/mssql/log/errorlog ]; then
+          echo "/var/opt/mssql/log/errorlog"
+        else
+          echo ""
+        fi
+      changed_when: false
+      register: __mssql_errorlog
+
+    - name: Check if the set edition matches the existing edition
+      shell: |
+        errorlog_edition="$(grep -oi '{{ __verify_mssql_edition }} edition' \
+        {{ __mssql_errorlog.stdout }})"
+        if [ -z "${errorlog_edition}" ]; then
+          edition_matches=false
+        else
+          edition_matches=true
+        fi
+        echo "${edition_matches}"
+      register: __mssql_edition_matches
+      ignore_errors: true
+      changed_when: false
+
+    - name: Verify if the edition matches
+      assert:
+        that: __mssql_edition_matches.stdout | bool
+
+- name: Verify the IP address setting
+  when: __mssql_ip_address_matches is defined
+  block:
+    - name: Check if the IP address setting is defined
+      command: /opt/mssql/bin/mssql-conf get network ipaddress
+      ignore_errors: true
+      changed_when: false
+      register: __mssql_conf_get_ip_address
+
+    - name: Verify if the IP address setting matches
+      assert:
+        that: >-
+          __mssql_conf_get_ip_address.stdout | regex_search('[0-9].*') ==
+          __mssql_ip_address_matches
+
+- name: Verify the TCP port setting
+  when: __mssql_tcp_port_matches is defined
+  block:
+    - name: Check if the TCP port setting is defined
+      command: /opt/mssql/bin/mssql-conf get network tcpport
+      ignore_errors: true
+      changed_when: false
+      register: __mssql_conf_get_tcp_port
+
+    - name: Verify if the TCP port setting matches
+      assert:
+        that: >-
+          __mssql_conf_get_tcp_port.stdout | regex_search('[0-9].*') ==
+          __mssql_tcp_port_matches
+
+- name: Verify password
+  when: __verify_mssql_password is defined
+  block:
+    - name: Log in using the set password
+      include_tasks: ../../tasks/verify_password.yml
```

Comment:
I see - it passes back the result using `__verify_mssql_password` - so in addition to using `include_role` and `tasks_from` here, you'll also have to use `public: true` so that you can use the private role variables outside of the role

Reply:
You are correct.

Reply:
Done that. I faced a weird issue with the evaluation of variables being defined or not. Ansible worked fine when mssql_password was defined:
```
TASK [linux-system-roles.mssql : debug] ******************************************************************************************************************************************************
ok: [127.0.0.3] => {
    "mssql_password is defined": "true!"
}
```
But when mssql_password was undefined, it returned `VARIABLE IS NOT DEFINED` instead of just `false`:
```
TASK [linux-system-roles.mssql : debug] ******************************************************************************************************************************************************
ok: [127.0.0.3] => {
    "mssql_password is defined": "VARIABLE IS NOT DEFINED!"
}
```
This was caused by setting the var like `mssql_password: "{{ __verify_mssql_password }}"`. This confused Ansible, so I introduced another variable `__mssql_password` for the same purpose.